docs/reference/files/file-libname-py.rst

```diff
@@ -0,0 +1,334 @@
+
+``LIBAPI``
+^^^^^^^^^^
+
+**Status:** Required
+
+**Purpose:** Declares the API version of this charm library.
+
+**Type:** ``int``
+
+**Value:** ``LIBAPI``` is set to an initial state of ``0``. In general,
+``LIBAPI`` must match the major version in the import path.
+
+``LIBPATCH``
+^^^^^^^^^^^^
+
+**Status:** Required
+
+**Purpose:** Declares the patch version of this charm library.
+
+**Type:** ``int``
+
+**Value:** ``LIBPATCH`` is set to an initial state of ``1``. In general, it must
+match the current patch version (needs to be updated when changing).
+
+.. note::
+
+    While ``LIBPATCH`` can be set to ``0``, it is not allowed to set both ``LIBAPI``
+    and ``LIBPATCH`` to ``0``. As such, a charm lib may have a version ``0.1`` and
+    a version ``1.0``, but not a version ``0.0``.
+
+``PYDEPS``
+^^^^^^^^^^
+
+**Status:** Optional
+
+**Purpose:** Declares external Python dependencies for the library.
+
+When using the ``charm`` plugin, Charmcraft will make sure to install them in the
+virtual environment created in any charm that includes the library.
+
+**Type:** ``list[str]``
+
+Each string is a regular "pip installable" Python dependency that will be retrieved
+from PyPI in the usual way (subject to the user's system configuration) and which
+supports all dependency formats (just the package name, a link to a Github project,
+etc.).
+
+.. dropdown:: Examples
+
+    .. code:: python
+
+        PYDEPS=["jinja2"]
+
+    .. code:: python
+
+        PYDEPS = ["pyyaml", "httpcore<0.15.0,>=0.14.5"]
+
+    .. code:: python
+
+        PYDEPS = [
+            "git+https://github.com/canonical/operator/#egg=ops",
+            "httpcore<0.15.0,>=0.14.5",
+            "requests",
+        ]
+
+Note that when called to install all the dependencies from the charm and all the
+used libraries, ``pip`` may detect conflicts between the requested packages and
+their versions. This is a feature, because it's always better to detect
+incompatibilities between dependencies at this moment than when the charm is being
+deployed or run after deployment.
+
+Code
+^^^^
+
+After the docstring and the metadata, there's the library code.
+This is regular Python code.
+
+Popular libraries
+-----------------
+
+This is a list of some popular charm libraries available from Charmhub.
+
+.. note::
+
+    This list does not and will not contain all charm libraries on Charmhub. However if
+    you believe a library is missing from this list, please
+    `open a pull request <https://github.com/canonical/charmcraft/pull/new/>`_ adding
+    the library you believe to be missing.
+
+Libraries that define relations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following libraries provide programmatic instructions for relating to a specific
+charm.
+
+.. list-table::
+    :header-rows: 1
+
+    * - Library
+      - Used in
+      - Description
+    * - `fluentbit <https://charmhub.io/fluentbit/libraries/fluentbit>`_
+      - - `fluentbit <https://charmhub.io/fluentbit>`_
```

Comment:
```suggestion
      - `fluentbit <https://charmhub.io/fluentbit>`_
```

A handful of column 2 cells in this block have single-item lists, whereas in the original text they were regular paragraphs.